src/TRestAxionTrueWolterOptics.cxx

```diff
@@ -174,6 +174,10 @@ void TRestAxionTrueWolterOptics::Initialize() {
     fAlpha = GetAlpha();
     fThickness = GetThickness();
 
+    fXSep.clear();
+    for (const auto& a : fAlpha)
+        fXSep.push_back(2 * (fR1 - fR3 - fMirrorLength * TMath::Sin(a)) / TMath::Tan(a));
```

Comment:
`fR3` is a `std::vector` that's why the compilation is failing, and validation is not going through.

Your need to access the elements of `fR1` and `fR3`.

```
for( size_t n = 0; n < fAlpha.size(); n++ )
     fXSep.push_back(2 * (fR1[n] - fR3[n] - fMirrorLength * TMath::Sin(fAlpha[n])) / TMath::Tan(fAlpha[n]));
```

where I am assuming `fAlpha`, `fR1` and `fR3` have the same number of elements.

Reply:
Ah, yeah, that's true!